container_pipeline/lib/openshift.py

```diff
@@ -0,0 +1,132 @@
+import logging
+import subprocess
+import time
+
+from container_pipeline.lib import settings
+from container_pipeline.lib.command import run_cmd
+
+
+class OpenshiftError(Exception):
+    pass
+
+
+class Openshift(object):
+
+    def __init__(self, endpoint=None, user=None, password=None,
+                 config=None, cert=None, logger=None):
+        self.endpoint = endpoint or settings.OPENSHIFT_ENDPOINT
+        self.user = user or settings.OPENSHIFT_USER
+        self.password = password or settings.OPENSHIFT_PASSWORD
+        self.config = config or settings.OC_CONFIG
+        self.cert = cert or settings.OC_CERT
+        self.oc_cmd_suffix = (
+            '--config {config} --certificate-authority {cert}'.format(
+                config=self.config, cert=self.cert))
+        self.logger = logger or logging.getLogger('console')
+
+    def login(self, user=None, password=None):
+        """Login to openshift"""
+        user = user or self.user
+        password = password or self.password
+        try:
+            self.logger.debug(
+                run_cmd(
+                    'oc login {endpoint} -u {user} -p {password} {suffix}'
+                    .format(
+                        endpoint=self.endpoint, user=user,
+                        password=password, suffix=self.oc_cmd_suffix
+                    ))
+            )
+        except subprocess.CalledProcessError as e:
+            raise OpenshiftError(
+                'Openshift login error: {}'.format(e))
+
+    def use_project(self, project):
+        """Use an openshift project"""
+        self.logger.debug('Using openshift project: {}'.format(project))
+        try:
+            self.logger.debug(
+                'oc project {project} {suffix}'.format(
+                    project=project, suffix=self.oc_cmd_suffix))
+        except subprocess.CalledProcessError as e:
+            raise OpenshiftError(
+                'Unable to use openshift project {}: {}'.format(project, e))
+
+    def build(self, project, build):
+        """Run build for a project"""
+        self.logger.debug('Run openshift project build: {}/{}'
+                          .format(build, project))
+        try:
+            output = run_cmd(
+                'oc --namespace {project} start-build {build} '
+                '{suffix}'.format(
+                    project=project, build=build, suffix=self.oc_cmd_suffix)
+            )
+            build_id = output.split('"')[-1].rstrip()
```

Comment:
@dharmit I will document this once I have a working development instance. Devcloud is down. @bamachrn if you could add the expected output in comments above this line, it'd be great.

@bamachrn one question. If the output is ``Build-1``, ``Test-1``, why are we splitting with `"`?